jolpica_api/ergastapi/serializers.py

```diff
@@ -118,6 +106,44 @@ def get_qualifying(self, race: Round):
     def get_sprint(self, race: Round):
         return self.get_session_date_time(race, SessionType.SPRINT_RACE)
 
+    def get_sprint_qualifying(self, race: Round):
+        return self.get_session_date_time(race, SessionType.SPRINT_QUALIFYING1)
+
+    class Meta:
+        model = Round
+        fields = ["season", "round", "url", "raceName", "Circuit", "date", "time"]
+
+
+class ListRaceSerializer(serializers.ListSerializer):
+    def to_representation(self, data: list[Round]) -> dict:
+        races = list()
+        for round_ in data:
+            # check if there is any kind of sprint sesssion and if yes, use the sprint weekend format for that year
+            if round_.sessions.filter(type__startswith="S").exists():
+                if round_.season.year in (2021, 2022):
+                    session_fields = ["FirstPractice", "SecondPractice", "Qualifying", "Sprint"]
+                elif round_.season.year == 2023:
+                    session_fields = ["FirstPractice", "Qualifying", "Sprint", "SprintShootout"]
+                elif round_.season.year >= 2024:
+                    session_fields = ["FirstPractice", "Qualifying", "Sprint", "SprintQualifying"]
+            else:
+                session_fields = ["FirstPractice", "SecondPractice", "ThirdPractice", "Qualifying"]
+
+            races.append(RaceSerializer(session_fields=session_fields).to_representation(round_))
+
+        return races
+
+
+class RaceSerializer(BaseRaceSerializer):
+    FirstPractice = serializers.SerializerMethodField(method_name="get_first_practice")
+    SecondPractice = serializers.SerializerMethodField(method_name="get_second_practice")
+    ThirdPractice = serializers.SerializerMethodField(method_name="get_third_practice")
+    Qualifying = serializers.SerializerMethodField(method_name="get_qualifying")
+    Sprint = serializers.SerializerMethodField(method_name="get_sprint")
+    SprintQualifying = serializers.SerializerMethodField(method_name="get_sprint_qualifying")
+    # Sprint shooutout is a qualifying session. We can reuse the same method because it's the same session type.
+    SprintShootout = serializers.SerializerMethodField(method_name="get_sprint_qualifying")
```

Comment:
Does this mean the value of both of these fields will always be identical?

Reply:
Sprint Shootout is the exact same type of session as Sprint Qualifying. But in 2023 it was called "Sprint Shootout". Then they renamed it to "Sprint Qualifying" for 2024 (and moved it from the 3rd to the 2nd session). 
The value of these two field will always be identical, but the further logic for dynamic fields will always remove at least one of them before serialization. Because it is just a naming thing here. In 2021 and 2022, we want none of these two. In 2023, we want the first one and in 2024+ we want the second one. Because I can easily remove fields dynamically, I just define both and then remove what isn't needed in the __init__.